### PR TITLE
default #[trace] to fn name

### PR DESCRIPTION
**Pull request by jon-chuang**

### CODE REVIEW

File: minitrace-macro/src/lib.rs [hunk 1]
```diff
@@ -15,23 +15,27 @@ use proc_macro2::TokenStream;
...
 
 struct Args {
     event: String,
     enter_on_poll: bool,
 }
 
 impl Args {
-    fn parse(input: AttributeArgs) -> Args {
+    fn parse(default_name: String, input: AttributeArgs) -> Args {
+        let mut next = input.get(1);
```
Comment (zhongzc) [inline minitrace-macro/src/lib.rs:27]:
```suggestion
        let (name, next) = match input.get(0) {
            Some(arg0) => match arg0 {
                NestedMeta::Lit(Lit::Str(name)) => (name.value(), input.get(1)),
                _ => (default_name, input.get(0)),
            },
            None => (default_name, None),
        };
```

**Comment by zhongzc** [comment]
Can you please follow https://github.com/tikv/minitrace-rust/pull/95/checks?check_run_id=5299855052 to sign off your commits?

**Review by zhongzc — APPROVED** [review]

**Comment by jon-chuang** [comment]
apologies, will take note in the future

**Comment by andylokandy** [comment]
Why it's necessary to [insert lifetimes before generics](https://github.com/tikv/minitrace-rust/pull/95/commits/fdedcf63f2ebfa66fae6dee48059963f2aa778d3)? Would you like to add a test showing the case?

**Comment by jon-chuang** [comment]
@andylokandy it fails to compile otherwise. 

Check rust book: https://doc.rust-lang.org/reference/items/generics.html

**Comment by andylokandy** [comment]
@jon-chuang Did it fail before this PR? I mean, why did not we catch that error in the unit test?

**Comment by jon-chuang** [comment]
Yes, there could be more tests if required. But it cannot be caught in unit test as compile will simply fail. But it could be caught in CI if you set that up.

**Comment by andylokandy** [comment]
Would you mind giving me an example that will fail to compile before this PR? So that I can add a regression test for it.

**Pull request merged**